README.md
ColorFilter generator and presets to use with ColorFiltered widget.

## Installation

First, add `colorfilter_generator:` as a [dependency in your pubspec.yaml file](https://flutter.io/platform-plugins/).
  

Presets included:   AddictiveBlue, AddictiveRed, Aden, Amaro, Ashby, Brannan, Brooklyn, Charmes, Clarendon, Crema, Dogpatch, Earlybird, F1977, Gingham, Ginza, Hefe, Helena, Hudson, Inkwell, Juno, Kelvin, Lark, LoFi, Ludwig, Maven, Mayfair, Moon, Nashville, Perpetua, Reyes, Rise, Sierra, Skyline, Slumber, Stinson, Sutro, Toaster, Valencia, Vesper, Walden, Willow, XProII
  
  
## Usage  
  
```dart
ColorFiltered(
  colorFilter: ColorFilter.matrix(
    ColorFilterAddons.brightness(0.4),
  ),
  child: Image.network('https://example.com/path/to/image.png'),
)
```

## Custom Filter  
  
```dart
ColorFilterGenerator myFilter = ColorFilterGenerator(
  name: "CustomFilter",
  filters: [
    ColorFilterAddons.brightness(.1),
    ColorFilterAddons.contrast(.1),
    ColorFilterAddons.saturation(.15),
    PresetFilters.ludwig,
    PresetFilters.ludwig.opacity(0.20),
  ]
);

// Usage
ColorFiltered(
  colorFilter: ColorFilter.matrix(myFilter.matrix),
  child: Image.network('https://example.com/path/to/image.png'),
)
```

CHANGELOG.md
## 0.0.8

* Implemented build method to apply filter to given child

## 0.0.7

* Added github repository.

## 0.0.6

* Optimization

## 0.0.5

* Bug fixes

## 0.0.4

* Added opacity method to filter

## 0.0.3

* Added matrix property

## 0.0.2

* Transfered presets to presets.dart

## 0.0.1

* Initial version


lib/addons.dart
import 'dart:math';

// | R' |   | a00 a01 a02 a03 a04 |   | R |
// | G' |   | a10 a11 a22 a33 a44 |   | G |
// | B' | = | a20 a21 a22 a33 a44 | * | B |
// | A' |   | a30 a31 a22 a33 a44 |   | A |
// | 1  |   |  0   0   0   0   1  |   | 1 |
class ColorFilterAddons {
  // RGBA colorOverlay(RGBA color, num red, num green, num blue, num scale) {
  //   return new RGBA(
  //     red: (color.red - (color.red - red) * scale),
  //     green: (color.green - (color.green - green) * scale),
  //     blue: (color.blue - (color.blue - blue) * scale),
  //     alpha: color.alpha,
  //   );
  // }
  /// color(1 - scale) - color * scale
  static List<double> colorOverlay(
      double red, double green, double blue, double scale) {
    return [
      (1 - scale),
      0,
      0,
      0,
      -1 * red * scale,
      0,
      (1 - scale),
      0,
      0,
      -1 * green * scale,
      0,
      0,
      (1 - scale),
      0,
      -1 * blue * scale,
      0,
      0,
      0,
      1,
      0
    ];
  }

  static List<double> rgbScale(double r, double g, double b) {
    return [
      r,
      0,
      0,
      0,
      0,
      0,
      g,
      0,
      0,
      0,
      0,
      0,
      b,
      0,
      0,
      0,
      0,
      0,
      1,
      0,
    ];
  }

  static List<double> addictiveColor(double r, double g, double b) {
    return [
      1,
      0,
      0,
      0,
      r,
      0,
      1,
      0,
      0,
      g,
      0,
      0,
      1,
      0,
      b,
      0,
      0,
      0,
      1,
      0,
    ];
  }

  /// (0.2126 * color.red + 0.7152 * color.green + 0.0722 * color.blue)
  static List<double> grayscale() {
    return [
      0.2126,
      0.7152,
      0.0722,
      0,
      0,
      0.2126,
      0.7152,
      0.0722,
      0,
      0,
      0.2126,
      0.7152,
      0.0722,
      0,
      0,
      0,
      0,
      0,
      1,
      0,
    ];
  }

  // red: ((r * (1 - (0.607 * adj))) + (g * .769 * adj) + (b * .189 * adj)),
  // green: ((r * .349 * adj) + (g * (1 - (0.314 * adj))) + (b * .168 * adj)),
  // blue: ((r * .272 * adj) + (g * .534 * adj) + (b * (1 - (0.869 * adj)))),
  // alpha: color.alpha
  static List<double> sepia(double value) {
    return [
      (1 - (0.607 * value)),
      0.769 * value,
      0.189 * value,
      0,
      0,
      0.349 * value,
      (1 - (0.314 * value)),
      0.168 * value,
      0,
      0,
      0.272 * value,
      0.534 * value,
      (1 - (0.869 * value)),
      0,
      0,
      0,
      0,
      0,
      1,
      0,
    ];
  }

  /// Invert the colors
  static List<double> invert() {
    return [
      -1,
      0,
      0,
      0,
      255,
      0,
      -1,
      0,
      0,
      255,
      0,
      0,
      -1,
      0,
      255,
      0,
      0,
      0,
      1,
      0,
    ];
  }

  /// Brightness adjustment
  static List<double> brightness(double value) {
    if (value <= 0) {
      value = value * 255;
    } else {
      value = value * 100;
    }

    if (value == 0) {
      return [
        1,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
      ];
    }

    return List<double>.from(<double>[
      1,
      0,
      0,
      0,
      value,
      0,
      1,
      0,
      0,
      value,
      0,
      0,
      1,
      0,
      value,
      0,
      0,
      0,
      1,
      0
    ]).map((i) => i.toDouble()).toList();
  }

  /// Contrast adjustment
  static List<double> contrast(double value) {
    // RGBA contrast(RGBA color, num adj) {
    //   adj *= 255;
    //   double factor = (259 * (adj + 255)) / (255 * (259 - adj));
    //   return new RGBA(
    //     red: (factor * (color.red - 128) + 128),
    //     green: (factor * (color.green - 128) + 128),
    //     blue: (factor * (color.blue - 128) + 128),
    //     alpha: color.alpha,
    //   );
    // }
    double adj = value * 255;
    double factor = (259 * (adj + 255)) / (255 * (259 - adj));

    return [
      factor,
      0,
      0,
      0,
      128 * (1 - factor),
      0,
      factor,
      0,
      0,
      128 * (1 - factor),
      0,
      0,
      factor,
      0,
      128 * (1 - factor),
      0,
      0,
      0,
      1,
      0,
    ];
  }

  /// Hue adjustment
  static List<double> hue(double value) {
    value = value * pi;

    if (value == 0) {
      return [
        1,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
      ];
    }

    var cosVal = cos(value);
    var sinVal = sin(value);
    var lumR = 0.213;
    var lumG = 0.715;
    var lumB = 0.072;

    return List<double>.from(<double>[
      (lumR + (cosVal * (1 - lumR))) + (sinVal * (-lumR)),
      (lumG + (cosVal * (-lumG))) + (sinVal * (-lumG)),
      (lumB + (cosVal * (-lumB))) + (sinVal * (1 - lumB)),
      0,
      0,
      (lumR + (cosVal * (-lumR))) + (sinVal * 0.143),
      (lumG + (cosVal * (1 - lumG))) + (sinVal * 0.14),
      (lumB + (cosVal * (-lumB))) + (sinVal * (-0.283)),
      0,
      0,
      (lumR + (cosVal * (-lumR))) + (sinVal * (-(1 - lumR))),
      (lumG + (cosVal * (-lumG))) + (sinVal * lumG),
      (lumB + (cosVal * (1 - lumB))) + (sinVal * lumB),
      0,
      0,
      0,
      0,
      0,
      1,
      0,
    ]).map((i) => i.toDouble()).toList();
  }

  /// Saturation adjustment
  static List<double> saturation(double value) {
    value = value * 100;

    if (value == 0) {
      return [
        1,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
      ];
    }

    var x =
        ((1 + ((value > 0) ? ((3 * value) / 100) : (value / 100)))).toDouble();
    var lumR = 0.3086;
    var lumG = 0.6094;
    var lumB = 0.082;

    return List<double>.from(<double>[
      (lumR * (1 - x)) + x,
      lumG * (1 - x),
      lumB * (1 - x),
      0,
      0,
      lumR * (1 - x),
      (lumG * (1 - x)) + x,
      lumB * (1 - x),
      0,
      0,
      lumR * (1 - x),
      lumG * (1 - x),
      (lumB * (1 - x)) + x,
      0,
      0,
      0,
      0,
      0,
      1,
      0,
    ]).map((i) => i.toDouble()).toList();
  }
}


example/main.dart
import 'package:flutter/material.dart';
import 'package:colorfilter_generator/addons.dart';

void main() {
  runApp(
    const MaterialApp(
      home: MyApp(),
    ),
  );
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('ColorFilter Example'),
      ),
      body: Center(
        child: ColorFiltered(
          colorFilter: ColorFilter.matrix(
            ColorFilterAddons.brightness(0.4),
          ),
          child: Column(
            children: [
              const Text('Some text with filter applied'),
              Image.network('https://example.com/path/to/image.png'),
            ],
          ),
        ),
      ),
    );
  }
}


lib/colorfilter_generator.dart
library colorfilter_generator;

import 'package:flutter/material.dart';
import 'package:matrix2d/matrix2d.dart';

/// The [ColorFilterGenerator] class to define a Filter which will applied to each color, consists of multiple [SubFilter]s
class ColorFilterGenerator {
  String name;
  List<List<double>> filters;
  List<double> matrix = [
    1,
    0,
    0,
    0,
    0,
    0,
    1,
    0,
    0,
    0,
    0,
    0,
    1,
    0,
    0,
    0,
    0,
    0,
    1,
    0
  ];

  ColorFilterGenerator({
    required this.name,
    required this.filters,
  }) {
    buildMatrix();
  }

  // | a00 a01 a02 a03 a04 |   | a00 a01 a02 a03 a04 |
  // | a10 a11 a22 a33 a44 |   | a10 a11 a22 a33 a44 |
  // | a20 a21 a22 a33 a44 | * | a20 a21 a22 a33 a44 |
  // | a30 a31 a22 a33 a44 |   | a30 a31 a22 a33 a44 |
  // |  0   0   0   0   1  |   |  0   0   0   0   1  |
  /// Build matrix of current filter
  void buildMatrix() {
    if (filters.isEmpty) {
      return;
    }

    Matrix2d m2d = const Matrix2d();

    List result = m2d.reshape([filters[0]], 4, 5);
    // List listA = m2d.reshape([filters[0]], 4, 5);

    for (int i = 1; i < filters.length; i++) {
      List listB = [
        ...(filters[i] is ColorFilterGenerator
            ? (filters[i] as ColorFilterGenerator).matrix
            : filters[i]),
        0,
        0,
        0,
        0,
        1,
      ];

      // print(listA);
      // print(listB);

      result = m2d.dot(
        result,
        m2d.reshape([listB], 5, 5),
      );
    }

    matrix = List<double>.from(result.flatten.sublist(0, 20));
  }

  /// Create new filter from this filter with given opacity
  ColorFilterGenerator opacity(double value) {
    return ColorFilterGenerator(name: name, filters: [
      ...filters,
      [
        value,
        0,
        0,
        0,
        0,
        0,
        value,
        0,
        0,
        0,
        0,
        0,
        value,
        0,
        0,
        0,
        0,
        0,
        1,
        0,
      ],
    ]);
  }

  /// Apply filter to given child
  Widget build(Widget child) {
    Widget tree = child;

    for (int i = 0; i < filters.length; i++) {
      tree = ColorFiltered(
        colorFilter: ColorFilter.matrix(filters[i]),
        child: tree,
      );
    }

    return tree;
  }
}


lib/presets.dart
import 'package:colorfilter_generator/addons.dart';
import 'package:colorfilter_generator/colorfilter_generator.dart';

/// Preset filters that can be used directly
class PresetFilters {
  /// NoFilter: No filter
  static ColorFilterGenerator none = ColorFilterGenerator(
    name: "No Filter",
    filters: [],
  );

  /// Clarendon: adds light to lighter areas and dark to darker areas
  static ColorFilterGenerator clarendon = ColorFilterGenerator(
    name: "Clarendon",
    filters: [
      ColorFilterAddons.brightness(.1),
      ColorFilterAddons.contrast(.1),
      ColorFilterAddons.saturation(.15),
    ],
  );

  /// Increase red color gradient
  static ColorFilterGenerator addictiveRed = ColorFilterGenerator(
    name: "AddictiveRed",
    filters: [
      ColorFilterAddons.addictiveColor(50, 0, 0),
    ],
  );

  /// Increase blue color gradient
  static ColorFilterGenerator addictiveBlue = ColorFilterGenerator(
    name: "AddictiveBlue",
    filters: [
      ColorFilterAddons.addictiveColor(0, 0, 50),
    ],
  );

  /// Gingham: Vintage-inspired, taking some color out
  static ColorFilterGenerator gingham = ColorFilterGenerator(
    name: "Gingham",
    filters: [
      ColorFilterAddons.sepia(.04),
      ColorFilterAddons.contrast(-.15),
    ],
  );

  /// Moon: B/W, increase brightness and decrease contrast
  static ColorFilterGenerator moon = ColorFilterGenerator(
    name: "Moon",
    filters: [
      ColorFilterAddons.grayscale(),
      ColorFilterAddons.contrast(-.04),
      ColorFilterAddons.brightness(0.1),
    ],
  );

  /// Lark: Brightens and intensifies colours but not red hues
  static ColorFilterGenerator lark = ColorFilterGenerator(
    name: "Lark",
    filters: [
      ColorFilterAddons.brightness(0.08),
      ColorFilterAddons.grayscale(),
      ColorFilterAddons.contrast(-.04),
    ],
  );

  /// Reyes: a vintage filter, gives your photos a “dusty” look
  static ColorFilterGenerator reyes = ColorFilterGenerator(
    name: "Reyes",
    filters: [
      ColorFilterAddons.sepia(0.4),
      ColorFilterAddons.brightness(0.13),
      ColorFilterAddons.contrast(-.05),
    ],
  );

  /// Juno: Brightens colors, and intensifies red and yellow hues
  static ColorFilterGenerator juno = ColorFilterGenerator(
    name: "Juno",
    filters: [
      ColorFilterAddons.rgbScale(1.01, 1.04, 1),
      ColorFilterAddons.saturation(0.3),
    ],
  );

  /// Slumber: Desaturates the image as well as adds haze for a retro, dreamy look – with an emphasis on blacks and blues
  static ColorFilterGenerator slumber = ColorFilterGenerator(
    name: "Slumber",
    filters: [
      ColorFilterAddons.brightness(.1),
      ColorFilterAddons.saturation(-0.5),
    ],
  );

  /// Crema: Adds a creamy look that both warms and cools the image
  static ColorFilterGenerator crema = ColorFilterGenerator(
    name: "Crema",
    filters: [
      ColorFilterAddons.rgbScale(1.04, 1, 1.02),
      ColorFilterAddons.saturation(-0.05),
    ],
  );

  /// Ludwig: A slight hint of desaturation that also enhances light
  static ColorFilterGenerator ludwig = ColorFilterGenerator(
    name: "Ludwig",
    filters: [
      ColorFilterAddons.brightness(.05),
      ColorFilterAddons.saturation(-0.03),
    ],
  );

  /// Aden: This filter gives a blue/pink natural look
  static ColorFilterGenerator aden = ColorFilterGenerator(
    name: "Aden",
    filters: [
      ColorFilterAddons.colorOverlay(228, 130, 225, 0.13),
      ColorFilterAddons.saturation(-0.2),
    ],
  );

  /// Perpetua: Adding a pastel look, this filter is ideal for portraits
  static ColorFilterGenerator perpetua = ColorFilterGenerator(
    name: "Perpetua",
    filters: [
      ColorFilterAddons.rgbScale(1.05, 1.1, 1),
    ],
  );

  /// Amaro: Adds light to an image, with the focus on the centre
  static ColorFilterGenerator amaro = ColorFilterGenerator(
    name: "Amaro",
    filters: [
      ColorFilterAddons.saturation(0.3),
      ColorFilterAddons.brightness(0.15),
    ],
  );

  /// Mayfair: Applies a warm pink tone, subtle vignetting to brighten the photograph center and a thin black border
  static ColorFilterGenerator mayfair = ColorFilterGenerator(
    name: "Mayfair",
    filters: [
      ColorFilterAddons.colorOverlay(230, 115, 108, 0.05),
      ColorFilterAddons.saturation(0.15),
    ],
  );

  /// Rise: Adds a "glow" to the image, with softer lighting of the subject
  static ColorFilterGenerator rise = ColorFilterGenerator(
    name: "Rise",
    filters: [
      ColorFilterAddons.colorOverlay(255, 170, 0, 0.1),
      ColorFilterAddons.brightness(0.09),
      ColorFilterAddons.saturation(0.1),
    ],
  );

  /// Hudson: Creates an "icy" illusion with heightened shadows, cool tint and dodged center
  static ColorFilterGenerator hudson = ColorFilterGenerator(
    name: "Hudson",
    filters: [
      ColorFilterAddons.rgbScale(1, 1, 1.25),
      ColorFilterAddons.contrast(0.1),
      ColorFilterAddons.brightness(0.15),
    ],
  );

  /// Valencia: Fades the image by increasing exposure and warming the colors, to give it an antique feel
  static ColorFilterGenerator valencia = ColorFilterGenerator(
    name: "Valencia",
    filters: [
      ColorFilterAddons.colorOverlay(255, 225, 80, 0.08),
      ColorFilterAddons.saturation(0.1),
      ColorFilterAddons.contrast(0.05),
    ],
  );

  /// X-Pro II: Increases color vibrance with a golden tint, high contrast and slight vignette added to the edges
  static ColorFilterGenerator xProII = ColorFilterGenerator(
    name: "X-Pro II",
    filters: [
      ColorFilterAddons.colorOverlay(255, 255, 0, 0.07),
      ColorFilterAddons.saturation(0.2),
      ColorFilterAddons.contrast(0.15),
    ],
  );

  /// Sierra: Gives a faded, softer look
  static ColorFilterGenerator sierra = ColorFilterGenerator(
    name: "Sierra",
    filters: [
      ColorFilterAddons.contrast(-0.15),
      ColorFilterAddons.saturation(0.1),
    ],
  );

  /// Willow: A monochromatic filter with subtle purple tones and a translucent white border
  static ColorFilterGenerator willow = ColorFilterGenerator(
    name: "Willow",
    filters: [
      ColorFilterAddons.grayscale(),
      ColorFilterAddons.colorOverlay(100, 28, 210, 0.03),
      ColorFilterAddons.brightness(0.1),
    ],
  );

  /// Lo-Fi: Enriches color and adds strong shadows through the use of saturation and "warming" the temperature
  static ColorFilterGenerator loFi = ColorFilterGenerator(
    name: "Lo-Fi",
    filters: [
      ColorFilterAddons.contrast(0.15),
      ColorFilterAddons.saturation(0.2),
    ],
  );

  /// Inkwell: Direct shift to black and white
  static ColorFilterGenerator inkwell = ColorFilterGenerator(
    name: "Inkwell",
    filters: [
      ColorFilterAddons.grayscale(),
    ],
  );

  /// Hefe: Hight contrast and saturation, with a similar effect to Lo-Fi but not quite as dramatic
  static ColorFilterGenerator hefe = ColorFilterGenerator(
    name: "Hefe",
    filters: [
      ColorFilterAddons.contrast(0.1),
      ColorFilterAddons.saturation(0.15),
    ],
  );

  /// Nashville: Warms the temperature, lowers contrast and increases exposure to give a light "pink" tint – making it feel "nostalgic"
  static ColorFilterGenerator nashville = ColorFilterGenerator(
    name: "Nashville",
    filters: [
      ColorFilterAddons.colorOverlay(220, 115, 188, 0.12),
      ColorFilterAddons.contrast(-0.05),
    ],
  );

  /// Stinson: washing out the colors ever so slightly
  static ColorFilterGenerator stinson = ColorFilterGenerator(
    name: "Stinson",
    filters: [
      ColorFilterAddons.brightness(0.1),
      ColorFilterAddons.sepia(0.3),
    ],
  );

  /// Vesper: adds a yellow tint that
  static ColorFilterGenerator vesper = ColorFilterGenerator(
    name: "Vesper",
    filters: [
      ColorFilterAddons.colorOverlay(255, 225, 0, 0.05),
      ColorFilterAddons.brightness(0.06),
      ColorFilterAddons.contrast(0.06),
    ],
  );

  /// Earlybird: Gives an older look with a sepia tint and warm temperature
  static ColorFilterGenerator earlybird = ColorFilterGenerator(
    name: "Earlybird",
    filters: [
      ColorFilterAddons.colorOverlay(255, 165, 40, 0.2),
      ColorFilterAddons.saturation(0.15),
    ],
  );

  /// Brannan: Increases contrast and exposure and adds a metallic tint
  static ColorFilterGenerator brannan = ColorFilterGenerator(
    name: "Brannan",
    filters: [
      ColorFilterAddons.contrast(0.2),
      ColorFilterAddons.colorOverlay(140, 10, 185, 0.1),
    ],
  );

  /// Sutro: Burns photo edges, increases highlights and shadows dramatically with a focus on purple and brown colors
  static ColorFilterGenerator sutro = ColorFilterGenerator(
    name: "Sutro",
    filters: [
      ColorFilterAddons.brightness(-0.1),
      ColorFilterAddons.saturation(-0.1),
    ],
  );

  /// Toaster: Ages the image by "burning" the centre and adds a dramatic vignette
  static ColorFilterGenerator toaster = ColorFilterGenerator(
    name: "Toaster",
    filters: [
      ColorFilterAddons.sepia(0.1),
      ColorFilterAddons.colorOverlay(255, 145, 0, 0.2),
    ],
  );

  /// Walden: Increases exposure and adds a yellow tint
  static ColorFilterGenerator walden = ColorFilterGenerator(
    name: "Walden",
    filters: [
      ColorFilterAddons.brightness(0.1),
      ColorFilterAddons.colorOverlay(255, 255, 0, 0.2),
    ],
  );

  /// 1977: The increased exposure with a red tint gives the photograph a rosy, brighter, faded look.
  static ColorFilterGenerator f1977 = ColorFilterGenerator(
    name: "1977",
    filters: [
      ColorFilterAddons.colorOverlay(255, 25, 0, 0.15),
      ColorFilterAddons.brightness(0.1),
    ],
  );

  /// Kelvin: Increases saturation and temperature to give it a radiant "glow"
  static ColorFilterGenerator kelvin = ColorFilterGenerator(
    name: "Kelvin",
    filters: [
      ColorFilterAddons.colorOverlay(255, 140, 0, 0.1),
      ColorFilterAddons.rgbScale(1.15, 1.05, 1),
      ColorFilterAddons.saturation(0.35),
    ],
  );

  /// Maven: darkens images, increases shadows, and adds a slightly yellow tint overal
  static ColorFilterGenerator maven = ColorFilterGenerator(
    name: "Maven",
    filters: [
      ColorFilterAddons.colorOverlay(225, 240, 0, 0.1),
      ColorFilterAddons.saturation(0.25),
      ColorFilterAddons.contrast(0.05),
    ],
  );

  /// Ginza: brightens and adds a warm glow
  static ColorFilterGenerator ginza = ColorFilterGenerator(
    name: "Ginza",
    filters: [
      ColorFilterAddons.sepia(0.06),
      ColorFilterAddons.brightness(0.1),
    ],
  );

  /// Skyline: brightens to the image pop
  static ColorFilterGenerator skyline = ColorFilterGenerator(
    name: "Skyline",
    filters: [
      ColorFilterAddons.saturation(0.35),
      ColorFilterAddons.brightness(0.1),
    ],
  );

  /// Dogpatch: increases the contrast, while washing out the lighter colors
  static ColorFilterGenerator dogpatch = ColorFilterGenerator(
    name: "Dogpatch",
    filters: [
      ColorFilterAddons.contrast(0.15),
      ColorFilterAddons.brightness(0.1),
    ],
  );

  /// Brooklyn
  static ColorFilterGenerator brooklyn = ColorFilterGenerator(
    name: "Brooklyn",
    filters: [
      ColorFilterAddons.colorOverlay(25, 240, 252, 0.05),
      ColorFilterAddons.sepia(0.3),
    ],
  );

  /// Helena: adds an orange and teal vibe
  static ColorFilterGenerator helena = ColorFilterGenerator(
    name: "Helena",
    filters: [
      ColorFilterAddons.colorOverlay(208, 208, 86, 0.2),
      ColorFilterAddons.contrast(0.15),
    ],
  );

  /// Ashby: gives images a great golden glow and a subtle vintage feel
  static ColorFilterGenerator ashby = ColorFilterGenerator(
    name: "Ashby",
    filters: [
      ColorFilterAddons.colorOverlay(255, 160, 25, 0.1),
      ColorFilterAddons.brightness(0.1),
    ],
  );

  /// Charmes: a high contrast filter, warming up colors in your image with a red tint
  static ColorFilterGenerator charmes = ColorFilterGenerator(
    name: "Charmes",
    filters: [
      ColorFilterAddons.colorOverlay(255, 50, 80, 0.12),
      ColorFilterAddons.contrast(0.05),
    ],
  );
}

/// List of filter presets
List<ColorFilterGenerator> presetFiltersList = [
  PresetFilters.none,
  PresetFilters.addictiveBlue,
  PresetFilters.addictiveRed,
  PresetFilters.aden,
  PresetFilters.amaro,
  PresetFilters.ashby,
  PresetFilters.brannan,
  PresetFilters.brooklyn,
  PresetFilters.charmes,
  PresetFilters.clarendon,
  PresetFilters.crema,
  PresetFilters.dogpatch,
  PresetFilters.earlybird,
  PresetFilters.f1977,
  PresetFilters.gingham,
  PresetFilters.ginza,
  PresetFilters.hefe,
  PresetFilters.helena,
  PresetFilters.hudson,
  PresetFilters.inkwell,
  PresetFilters.juno,
  PresetFilters.kelvin,
  PresetFilters.lark,
  PresetFilters.loFi,
  PresetFilters.ludwig,
  PresetFilters.maven,
  PresetFilters.mayfair,
  PresetFilters.moon,
  PresetFilters.nashville,
  PresetFilters.perpetua,
  PresetFilters.reyes,
  PresetFilters.rise,
  PresetFilters.sierra,
  PresetFilters.skyline,
  PresetFilters.slumber,
  PresetFilters.stinson,
  PresetFilters.sutro,
  PresetFilters.toaster,
  PresetFilters.valencia,
  PresetFilters.vesper,
  PresetFilters.walden,
  PresetFilters.willow,
  PresetFilters.xProII,
];
